From akpm@linux-foundation.org Mon Feb  5 16:16:48 2007
From: Grant Coady <grant_lkml@dodo.com.au>
Date: Mon, 05 Feb 2007 16:15:20 -0800
Subject: kernel/params.c: compile fail when CONFIG_SYSFS not set
To: greg@kroah.com
Cc: akpm@linux-foundation.org, grant_lkml@dodo.com.au, gcoady.lk@gmail.com
Message-ID: <200702060016.l160GNgo004035@shell0.pdx.osdl.net>


From: Grant Coady <grant_lkml@dodo.com.au>

kernel/profile.c:247: warning: 'profile_flip_buffers' defined but not used
kernel/profile.c:270: warning: 'profile_discard_flip_buffers' defined but not used
kernel/profile.c:337: warning: 'profile_cpu_callback' defined but not used
kernel/params.c:700: error: `module_uevent_ops' undeclared here (not in a function)
kernel/params.c:700: error: initializer element is not constant
kernel/params.c:700: error: (near initialization for `module_subsys.kset.uevent_ops')
kernel/params.c:700: error: initializer element is not constant
kernel/params.c:700: error: (near initialization for `module_subsys.kset')
make[1]: *** [kernel/params.o] Error 1
make: *** [kernel] Error 2

Signed-off-by: Grant Coady <gcoady.lk@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 kernel/params.c |    4 ++++
 1 file changed, 4 insertions(+)

--- gregkh-2.6.orig/kernel/params.c
+++ gregkh-2.6/kernel/params.c
@@ -691,6 +691,10 @@ static struct sysfs_ops module_sysfs_ops
 	.show = NULL,
 	.store = NULL,
 };
+
+static struct kset_uevent_ops module_uevent_ops = {
+	.filter = NULL,
+};
 #endif
 
 static struct kobj_type module_ktype = {
